appinventor/buildserver/src/com/google/appinventor/buildserver/util/Execution.java

```diff
@@ -29,6 +30,8 @@ public final class Execution {
   private static final Logger LOG = Logger.getLogger(Execution.class.getName());
   private static final Joiner joiner = Joiner.on(" ");
 
+  private static final int SUBPROCESS_TIMEOUT_SECONDS = 5 * 60; // 5 minutes
```

Comment:
I think a build is composed of something like 10 tasks, so if we wait 5 minutes for each a failing build could take almost an hour.

Reply:
Yeah, but things like D8 will definitely take more time than apksigner. Even D8 alone should take around 1-2 minutes maximum. The only reason why I added that much buffer is in case something goes really wrong, and we don't want to terminate too early the processes.

Unless we have data on what's the longest D8 can take on a legitimate, I wouldn't be confident setting a value lower than this. Also, this value varies depending on the CPU specs, hence if using a mix of different CPUs, each one may behave differently.  
Again, the reason why setting this to 5 minutes is as a mere guardrail preventing any hanging build in case we missed out anything else, not as a "global build timeout".

Reply:
Is it worth adding a global build timeout though? I can actually submit that change as a different PR. With this change, I'm just focusing on terminating all processes within 5 minutes at maximum, so if the buildserver is using a "not that good CPU", and D8 has a lot to do, it leaves buffer to not force exit by mistake.

Reply:
Most tasks are quick but some we know are slow. We may want to have a general timeout of 1 or 2 seconds and certain tasks like d8 can override it to a 1 minute or so. We also would want to make sure that any constraint would not prevent the companion from compiling when in CI or as part of a release cycle.

Reply:
> Is it worth adding a global build timeout though?

I have also added a global build timeout of 5 minutes. It should be more than enough, but inside the ProjectBuilder, the compiler thread now exits if the build did not complete within 5 minutes. Maybe it's too high, but I think it serves its purpose.